tests/numpy/test_stack.py

```diff
@@ -0,0 +1,84 @@
+# coding=utf-8
+# Copyright (C) 2020 NumS Development Team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import numpy as np
+
+
+# pylint: disable=import-outside-toplevel
+
+
+def test_atleast_1d(nps_app_inst):
+    import nums.numpy as nps
+    assert nps_app_inst is not None
+
+    x = 1.0
+    assert np.allclose(nps.atleast_1d(x).get(), np.atleast_1d(x))
+
+    x_np = np.arange(9).reshape(3, 3)
+    x_nps = nps.arange(9).reshape(3, 3)
+    assert np.allclose(nps.atleast_1d(x_np).get(), np.atleast_1d(x_np))
+    assert np.allclose(nps.atleast_1d(x_nps).get(), np.atleast_1d(x_np))
+    assert nps.atleast_1d(x_nps) is x_nps
+
+    assert np.allclose(nps.atleast_1d(1, [3, 4])[0].get(), np.atleast_1d(1, [3, 4])[0])
+    assert np.allclose(nps.atleast_1d(1, [3, 4])[1].get(), np.atleast_1d(1, [3, 4])[1])
+
+
+def test_atleast_2d(nps_app_inst):
+    import nums.numpy as nps
+    assert nps_app_inst is not None
+
+    x = 1.0
+    assert np.allclose(nps.atleast_2d(x).get(), np.atleast_2d(x))
+
+    x = np.arange(3.0)
+    assert np.allclose(nps.atleast_2d(x).get(), np.atleast_2d(x))
+
+    assert nps.atleast_2d(1, [1, 2], [[1, 2]])[0] == np.atleast_2d(1, [1, 2], [[1, 2]])[0]
+    assert nps.atleast_2d(1, [1, 2], [[1, 2]])[1] == np.atleast_2d(1, [1, 2], [[1, 2]])[1]
+    assert nps.atleast_2d(1, [1, 2], [[1, 2]])[2] == np.atleast_2d(1, [1, 2], [[1, 2]])[2]
+
+
+def test_atleast_3d(nps_app_inst):
```

Comment:
Looks like you test a scalar, a 1d, 2d, and 3d array passed to the function for scalars and lists (Python objects). From our discussion offline, there are 3 types we want to test: Python objects, numpy objects and BlockArray objects. In addition, it seems to properly test this functions behavior, we want to test a scalar, and data structures with 1, 2, 3, and 4 axes for each of these different types, and a few mixtures of those types as well. I think you can test these cases more easily by creating a function that generates the test inputs up to a certain dimension. Could you look into doing this for this test, as well as the other `atleast_nd` tests?

Reply:
Will do.